### ISSUE TITLE
WebSocketStream

### ISSUE THREAD

**Issue by ricea** [mozilla/standards-positions#970]
## Request for Mozilla Position on an Emerging Web Specification

* Specification title: WebSocketStream
* Specification or proposal URL (if available): https://whatpr.org/websockets/48.html#the-websocketstream-interface
* Explainer URL (if available): https://github.com/ricea/websocketstream-explainer/blob/master/README.md
* Proposal author(s) (`@`-mention GitHub accounts): @ricea 
* Caniuse.com URL (optional): 
* Bugzilla URL (optional): 
* Mozillians who can provide input (optional): 

### Other information

WebSocketStream is a new API for [the WebSocket protocol](https://datatracker.ietf.org/doc/html/rfc6455) that enables backpressure by integrating with Streams. The benefits are better behaviour when one side cannot keep up with the data rate, and a more modern API.

**Comment by valenting** [comment]
@jesup @jan-ivar Considering your experience with the WebTransport API do you have any thoughts regarding this proposal?

**Comment by smaug----** [comment]
@ricea the explainer talks about WebTransport as if it isn't a thing yet, but doesn't that have already multiple implementations. Is it not possible to support WebTransport API on top of various different kinds of network protocols?


(I don't personally have too strong opinions on WebSocketStream. The proposed API does look quite nice and simple when comparing to WebTransport which definitely is very complex)

@jesup @jan-ivar

**Comment by ricea** [comment]
> @ricea the explainer talks about WebTransport as if it isn't a thing yet, but doesn't that have already multiple implementations. Is it not possible to support WebTransport API on top of various different kinds of network protocols?

Yes, sorry, I missed that part when I was updating the explainer. I fixed it in https://github.com/ricea/websocketstream-explainer/pull/22.

I think [WebTransport over HTTP/2](https://datatracker.ietf.org/doc/draft-ietf-webtrans-http2/) will eventually be supported by browsers.

There is also a [polyfill of the WebTransport API over WebSocket](https://github.com/fails-components/webtransport?tab=readme-ov-file). Ironically, this would benefit from the backpressure support provided by WebSocketStream.
